tests/test_accelerated_op.py

```diff
@@ -0,0 +1,277 @@
+
+
+def test_adam_bp_non_accelerated():
+    from torchopt.accelerated_op._src import adam_op
+
+    backup = torchopt.accelerated_op.adam_op.adam_op
+    torchopt.accelerated_op.adam_op.adam_op = adam_op
+    AdamOp = torchopt.accelerated_op.adam_op.AdamOp
+
+    def forward_mu(
+        updates: torch.Tensor,
+        mu: torch.Tensor,
+        b1: float,
+    ) -> torch.Tensor:
+        """Adam forward mu."""
+        return mu.mul(b1).add(updates, alpha=1.0 - b1)
+
+    b1 = 0.2
+    updates_ref = torch.rand(5, 5, requires_grad=True)
+    mu_ref = torch.rand(5, 5, requires_grad=True)
+
+    new_mu_ref = forward_mu(updates_ref, mu_ref, b1)
+    new_mu_ref.sum().backward()
+
+    updates = updates_ref.clone().detach_().requires_grad_(True)
+    mu = mu_ref.clone().detach_().requires_grad_(True)
+    new_mu = AdamOp.MuOp.apply(updates, mu, b1)
+    new_mu.sum().backward()
+
+    assert torch.allclose(updates_ref.grad, updates.grad)
+    assert torch.allclose(mu_ref.grad, mu.grad)
+
+    def forward_nu(
+        updates: torch.Tensor,
+        nu: torch.Tensor,
+        b2: float,
+    ) -> torch.Tensor:
+        """Adam forward nu."""
+        return nu.mul(b2).add(updates.square(), alpha=1.0 - b2)
+
+    b2 = 0.2
+    updates_ref = torch.rand(5, 5, requires_grad=True)
+    nu_ref = torch.rand(5, 5, requires_grad=True)
+
+    new_nu_ref = forward_nu(updates_ref, nu_ref, b2)
+    new_nu_ref.sum().backward()
+
+    updates = updates_ref.clone().detach_().requires_grad_(True)
+    nu = nu_ref.clone().detach_().requires_grad_(True)
+    new_nu = AdamOp.NuOp.apply(updates, nu, b2)
+    new_nu.sum().backward()
+
+    assert torch.allclose(updates_ref.grad, updates.grad)
+    assert torch.allclose(nu_ref.grad, nu_ref.grad)
+
+    def forward_updates(
+        new_mu: torch.Tensor,
+        new_nu: torch.Tensor,
+        b1: float,
+        b2: float,
+        eps: float,
+        eps_root: float,
+        count: int,
+    ) -> torch.Tensor:
+        """Adam forward updates."""
+        inv_one_minus_pow_b1 = 1.0 / (1.0 - pow(b1, count))
+        inv_one_minus_pow_b2 = 1.0 / (1.0 - pow(b2, count) + eps_root)
+        return new_mu.mul(inv_one_minus_pow_b1).div(
+            new_nu.mul(inv_one_minus_pow_b2).add(eps).sqrt()
+        )
+
+    new_mu_ref.detach_().requires_grad_(True)
+    new_nu_ref.detach_().requires_grad_(True)
+    new_updates_ref = forward_updates(new_mu_ref, new_nu_ref, b1, b2, 0.0, 0.0, 1)
+    new_updates_ref.sum().backward()
+
+    new_mu.copy_(new_mu_ref).detach_().requires_grad_(True)
+    new_nu.copy_(new_nu_ref).detach_().requires_grad_(True)
+    new_updates = AdamOp.UpdatesOp.apply(new_mu, new_nu, (b1, b2, 0.0, 0.0, 1))
+    new_updates.sum().backward()
+
+    assert torch.allclose(new_mu_ref.grad, new_mu.grad)
+    assert torch.allclose(new_nu_ref.grad, new_nu.grad)
+
+    torchopt.accelerated_op.adam_op.adam_op = backup
+
+
+def test_adam_bp_accelerated():
+    if not torchopt.accelerated_op_available():
```

Comment:
Accelerated Op is always available for `cpu`. This function always fail because we do not have `cuda` device.

```suggestion
    if not CXX_ACCELERATED_OP_AVAILABLE:
```